src/dayview.ts

```diff
@@ -741,59 +453,99 @@ export class DayViewComponent implements ICalendarComponent, OnInit, OnChanges,
             len = eventSource ? eventSource.length : 0,
             startTime = this.range.startTime,
             endTime = this.range.endTime,
-            utcStartTime = Date.UTC(startTime.getFullYear(), startTime.getMonth(), startTime.getDate()),
-            utcEndTime = Date.UTC(endTime.getFullYear(), endTime.getMonth(), endTime.getDate()),
+            utcStartTime = Date.UTC(
+                startTime.getFullYear(),
+                startTime.getMonth(),
+                startTime.getDate()
+            ),
+            utcEndTime = Date.UTC(
+                endTime.getFullYear(),
+                endTime.getMonth(),
+                endTime.getDate()
+            ),
             currentViewIndex = this.currentViewIndex,
             rows = this.views[currentViewIndex].rows,
-            allDayEvents: IDisplayAllDayEvent[] = this.views[currentViewIndex].allDayEvents = [],
+            allDayEvents: IDisplayAllDayEvent[] = (this.views[
+                currentViewIndex
+            ].allDayEvents = []),
             oneHour = 3600000,
             eps = 0.016,
             rangeStartRowIndex = this.startHour * this.hourSegments,
             rangeEndRowIndex = this.endHour * this.hourSegments;
+        const allCategories = this.extractAllCategories(eventSource);
```

Comment:
this extractAllCategories is a heavy operation as need to loop the whole eventSource. if should be only used when dayviewShowCategoryView is true